ceph/rbd/workflows/rbd_trash.py

```diff
@@ -0,0 +1,211 @@
+        return False
+
+    images_in_trash = [trash_image.get("name") for trash_image in trash_list]
+    if image in images_in_trash:
+        log.info(f"Image {pool}/{image} is moved to trash successfully")
+        return True
+    else:
+        log.error(f"Image {pool}/{image} is not present in trash")
+        if test_ops_parallely:
+            raise Exception(f"Image {pool}/{image} is not present in trash")
+        return False
+
+
+def move_image_to_trash_and_verify(**kw):
+    """
+    Move the specified image to trash and verify that
+    the image is present in trash
+    kw:{
+        "rbd": <>,
+        "pool": <>,
+        "image": <>,
+        "image_conf": <>,
+        "test_ops_parallely": <>
+    }
+    """
+    rbd = kw.get("rbd")
+    pool = kw.get("pool")
+    image = kw.get("image")
+    test_ops_parallely = kw.get("test_ops_parallely", False)
+    image_conf = kw.get("image_conf", {})
+    expires_at = image_conf.get("expires-at", None)
+
+    log.info(f"Moving image {pool}/{image} to trash")
+    trash_conf = {"pool": pool, "image": image}
+    if expires_at:
+        trash_conf.update({"expires-at": f'"{expires_at}"'})
+    out, err = rbd.trash.mv(**trash_conf)
+    if (out or err) and f"image {image} will expire at" not in out + err:
+        log.error(f"Moving image {pool}/{image} to trash failed with err {out} {err}")
+        if test_ops_parallely:
+            raise Exception(
+                f"Moving image {pool}/{image} to trash failed with err {out} {err}"
+            )
+        return 1
+
+    log.info(f"Verifying whether image {pool}/{image} is present in trash")
+
+    if is_image_present_in_trash(rbd=rbd, pool=pool, image=image):
+        log.info(f"Image {pool}/{image} is present in trash")
+    else:
+        log.error(f"Image {pool}/{image} not present in trash")
+        if test_ops_parallely:
+            raise Exception(f"Image {pool}/{image} not present in trash")
+        return 1
+
+    return 0
+
+
+def restore_image_from_trash_and_verify(**kw):
+    """
+    Restore the specified image from trash and verify that
+    the image is not present in trash instead present in rbd ls
+    kw:{
+        "rbd": <>,
+        "pool": <>,
+        "image": <>,
+        "image_conf": <>,
+        "test_ops_parallely": <>
+    }
+    """
+    rbd = kw.get("rbd")
+    pool = kw.get("pool")
+    image = kw.get("image")
+    test_ops_parallely = kw.get("test_ops_parallely", False)
+
+    log.info(f"Get image id for {pool}/{image} from trash list")
+
+    out, err = rbd.trash.ls(pool=pool, format="json")
+    if err:
+        log.error(f"Listing images in trash failed for pool {pool} with err {err}")
+        if test_ops_parallely:
+            raise Exception(
+                f"Listing images in trash failed for pool {pool} with err {err}"
+            )
+        return 1
+
+    trash_list = json.loads(out)
+    image_id = [
+        trash_image.get("id")
```

Comment:
If there is an image, there will definitely be an ID. So I think we don't need a nonetype check here. Please let me know if you still think otherwise.